src/components/app.tsx

```diff
@@ -5,13 +5,41 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import { Dispatch } from 'redux';
+import { StateWithHistory } from 'redux-undo';
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+import { datasetUrlLoad } from '../actions';
+import { StateBase } from '../models/index';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
 
-class AppBase extends React.PureComponent<any, {}> {
+export type VoyagerConfig = Object;
+export type VoyagerData = Data;
+
+interface Props extends React.Props<AppBase> {
+  config?: VoyagerConfig;
+  data?: Data;
+  dispatch: Dispatch<StateWithHistory<Readonly<StateBase>>>;
+}
+
+class AppBase extends React.PureComponent<Props, {}> {
+
+  constructor(props: any) {
+    super(props);
+  }
+
+  public componentWillUpdate(nextProps: Props) {
+    this.update(nextProps);
```

Comment:
PureComponent provides an implementation that shallow compares with previous props. At this point I think that should be sufficient. We can override that implementation later down the road if necessary, though i don't think we will need to given how fundamental data and config are to what is rendered.